### DISCUSSION TITLE
Usability of 3D Transform Gizmo in local mode with multiple selection

### DISCUSSION THREAD

**Discussion by CookieBadger** [godotengine/godot-proposals#6202] [3D]
With https://github.com/godotengine/godot/pull/53684 the behavior of 3D transform gizmos changed. I'd like to revive discussion on the Gizmo's behavior with multiple selection with local transform. It is especially unintuive when selecting several objects with the same rotation (see gif) 
![MultipleSelection](https://user-images.githubusercontent.com/53947784/215357586-221f3c21-c076-4378-8d05-79c0e2034d7e.gif)
There is probably good reason to do it this way, but off the top of my head, I would expect the gizmo would align with the average rotation of the nodes, so it at least points in the right directions when selected nodes all have the same rotation. The other issue is that visually, the gizmo does not change at all when switching between local and global mode, thus often resulting in eyebrow-raising surprises, when objects start moving in random directions. I'm not confident enough with this to make either a proposal or an issue, but it did bog my mind a few times during level-editing. One possibility could be, that each selected node could show its own tiny gizmo that shows what their local direction of the axis it is moved on would be, but that again would only really work for moving and might result in undesired clutter. Like this:
![image](https://user-images.githubusercontent.com/53947784/215358732-8e2b7170-5d26-4c2f-89af-31592f2f355b.png)

Also, afaik, there still is no documentation on the multiple selection gizmo, as discussed in https://github.com/godotengine/godot/issues/68804...

Thoughts on this?

**Comment by essial** [comment]
At _minimum_ moving an object on an axis should move on the axis as represented by the axis widget.

As far as the group's "rotation", it should either be the average rotation of everything, rotation of the first selected object (if you select an object, then shift-select more), or worst case, on the default world orientation. Maybe even have this as an editor-level setting.

Just my opinion.

**Comment by SavoVuksan** [comment]
I also think that the gizmos should change orientation when in local mode. One of the behaviors that @essial said makes a lot of sense to me. If we look at blender and how they solve this issue they show the gizmo for the last selected object. If the transform mode is in local mode it transforms every object into their own local direction. if the movement mode is set to normal it moves all objects in the same local direction of the last selected object. This behavior makes a lot of sense to me.
# Blender Transform Modes
## Local Mode

https://user-images.githubusercontent.com/12395493/215726799-ef1de503-300b-458b-9f35-6388d9106ddf.mp4

## Normal Mode

https://user-images.githubusercontent.com/12395493/215726837-fe1e1a47-95bc-4eb6-8618-af5ee3db57aa.mp4
